src/flask/blueprints.py

```diff
@@ -284,11 +284,11 @@ def merge_dict_nested(self_dict, app_dict):
 
         if cli_resolved_group is None:
             app.cli.commands.update(self.cli.commands)
-        elif cli_resolved_group is _sentinel:
-            self.cli.name = self.name
-            app.cli.add_command(self.cli)
         else:
-            self.cli.name = cli_resolved_group
+            if cli_resolved_group is _sentinel:
```

Comment:
This change does not improve readability. Repeating a line once (or even twice) is perfectly fine; being overly DRY actually makes code harder to read here.